backend/core/src/application-flagged-sets/application-flagged-sets.service.ts

```diff
@@ -56,63 +66,87 @@ export class ApplicationFlaggedSetsService {
   }
 
   async resolve(dto: ApplicationFlaggedSetResolveDto) {
-    const afs = await this.afsRepository.findOne({
-      where: { id: dto.afsId },
-      relations: ["applications"],
-    })
-    if (!afs) {
-      throw new NotFoundException()
-    }
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-    afs.resolvingUser = this.request.user as User
-    afs.resolvedTime = new Date()
-    afs.status = FlaggedSetStatus.resolved
-    const appsToBeResolved = afs.applications.filter((afsApp) =>
-      dto.applications.map((appIdDto) => appIdDto.id).includes(afsApp.id)
-    )
-    const appsNotToBeResolved = afs.applications.filter(
-      (afsApp) => !dto.applications.map((appIdDto) => appIdDto.id).includes(afsApp.id)
-    )
+    return await getManager().transaction("SERIALIZABLE", async (transactionalEntityManager) => {
+      const transAfsRepository = transactionalEntityManager.getRepository(ApplicationFlaggedSet)
+      const transApplicationsRepository = transactionalEntityManager.getRepository(Application)
+      const afs = await transAfsRepository.findOne({
+        where: { id: dto.afsId },
+        relations: ["applications"],
+      })
+      if (!afs) {
+        throw new NotFoundException()
+      }
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      afs.resolvingUser = this.request.user as User
+      afs.resolvedTime = new Date()
+      afs.status = FlaggedSetStatus.resolved
+      const appsToBeResolved = afs.applications.filter((afsApp) =>
+        dto.applications.map((appIdDto) => appIdDto.id).includes(afsApp.id)
+      )
+      const appsNotToBeResolved = afs.applications.filter(
+        (afsApp) => !dto.applications.map((appIdDto) => appIdDto.id).includes(afsApp.id)
+      )
 
-    for (const appToBeResolved of appsToBeResolved) {
-      appToBeResolved.markedAsDuplicate = true
-      await this.applicationsRepository.save(appToBeResolved)
-    }
+      for (const appToBeResolved of appsToBeResolved) {
+        appToBeResolved.markedAsDuplicate = true
+        await transApplicationsRepository.save(appToBeResolved)
+      }
 
-    for (const appNotToBeResolved of appsNotToBeResolved) {
-      appNotToBeResolved.markedAsDuplicate = false
-      await this.applicationsRepository.save(appNotToBeResolved)
-    }
-    appsToBeResolved.forEach((app) => (app.markedAsDuplicate = true))
-    await this.afsRepository.save(afs)
-    return afs
+      for (const appNotToBeResolved of appsNotToBeResolved) {
+        appNotToBeResolved.markedAsDuplicate = false
+        await transApplicationsRepository.save(appNotToBeResolved)
+      }
+      appsToBeResolved.forEach((app) => (app.markedAsDuplicate = true))
+      await transAfsRepository.save(afs)
+      return afs
+    })
```

Comment:
I know this logic already existed, but these separate iterations could probably be condensed down into one. I'm not sure how large these record sets in this case get though, so not sure if it's worth any savings.